tiles/src/main/java/com/protomaps/basemap/layers/Boundaries.java

```diff
@@ -21,29 +21,213 @@ public String name() {
+        .setAttr("pmap:kind", kind)
+        .setAttr("pmap:kind_detail", kind_detail)
+        .setAttr("disputed", disputed)
+        .setBufferPixels(8);
+    }
+  }
+
   @Override
   public void processFeature(SourceFeature sf, FeatureCollector features) {
-    if (sf.canBeLine()) {
+    if (sf.canBeLine() && sf.hasTag("admin_level")) {
+      // Beware coastlines and coastal waters (eg with admin borders in large estuaries)
+      // like mouth of Columbia River between Oregon and Washington in USA
       if (sf.hasTag("natural", "coastline") || sf.hasTag("maritime", "yes")) {
         return;
       }
       List<OsmReader.RelationMember<AdminRecord>> recs = sf.relationInfo(AdminRecord.class);
       if (recs.size() > 0) {
         OptionalInt minAdminLevel = recs.stream().mapToInt(r -> r.relation().adminLevel).min();
         OptionalInt disputed = recs.stream().mapToInt(r -> r.relation().disputed).max();
-        var line =
-          features.line(this.name()).setId(FeatureId.create(sf)).setMinPixelSize(0).setAttr("pmap:min_admin_level",
-            minAdminLevel.getAsInt());
-        if (minAdminLevel.getAsInt() <= 2) {
-          line.setMinZoom(0);
-        } else if (minAdminLevel.getAsInt() <= 4) {
-          line.setMinZoom(3);
-        } else {
-          line.setMinZoom(10);
-        }
-
-        if (disputed.getAsInt() == 1) {
-          line.setAttr("disputed", 1);
+
+        var kind = "";
+        var kind_detail = "";
+
+        var min_zoom = 0;
+        var theme_min_zoom = 6;
+
+        // Core Tilezen schema properties
+        switch (minAdminLevel.getAsInt()) {
+          case 2 -> {
+            kind = "country";
+            kind_detail = "2";
+            // While country boundary lines should show up very early
+            min_zoom = 0;
+            // Natural Earth is used for low zooms (for compilation and tile size reasons)
+            theme_min_zoom = 6;
+          }
+          case 4 -> {
+            kind = "region";
+            kind_detail = "4";
+            // While region boundary lines should show up early-zooms
+            min_zoom = 6;
+            // Natural Earth is used for low zooms (for compilation and tile size reasons)
+            theme_min_zoom = 6;
+          }
+          case 6 -> {
+            kind = "county";
+            kind_detail = "6";
+            min_zoom = 8;
+            theme_min_zoom = 8;
+          }
+          case 8 -> {
+            kind = "locality";
+            kind_detail = "8";
+            min_zoom = 10;
+            theme_min_zoom = 10;
+          }
+        }
+
+        if (kind != "" && kind_detail != "") {
+          var line = features.line(this.name())
+            .setId(FeatureId.create(sf))
+            .setMinPixelSize(0)
+            .setAttr("pmap:min_admin_level", minAdminLevel.getAsInt())
+            .setAttr("pmap:kind", kind)
+            .setAttr("pmap:kind_detail", kind_detail)
+            .setAttr("pmap:min_zoom", min_zoom)
```

Comment:
So as is we will have a feature with `pmap:min_zoom=0` that starts appearing at 6, is this to enable tag continuity with the NE feature from zooms 0-5? Or used as a proxy for label weight/size?

Reply:
This is a bug... since names for text labels aren't included the min_zoom here should be omitted in v3.

To your question... Yes, to enable tag continuity with the NE features. Deliberate because to the end user they don't care where the feature came from... but it should have a consistent "min_zoom" based on when the user first saw it, not when it was transitioned to another data source on the server in the "black box".

Reply:
Fixed in https://github.com/protomaps/basemaps/pull/79.